packages/core-data/src/resolvers.js

```diff
@@ -173,6 +173,13 @@ export function* getEntityRecords( kind, name, query = {} ) {
 	}
 
 	yield receiveEntityRecords( kind, name, records, query );
+	for ( const record of records ) {
+		yield {
+			type: 'FINISH_RESOLUTION',
```

Comment:
These are meant to be low-level actions triggered by the framework itself, not sure we should be doing this here. Can we instead bail early in `getEntityRecord` if the record is available?

Reply:
@youknowriad this was my first hunch too, but then I realized that short-circuiting when the record is found will make it impossible to invalidate that record. The resolver will simply keep returning the cached value.

Reply:
how is that wrong?

Reply:
The more I think about it, the more I believe that `receiveEntityRecords` should somehow mark the records as resolved.  This proposed change goes one level higher which works because we don't have many places where records are received. But conceptually receiving a record means it no longer needs to be resolved, right? Maybe the resolvers part of the equation could somehow listen to `RECEIVE_ITEMS`?

Reply:
If you shortcircuit like I suggested above, the resolver will get called once, the bail early happens and the resolution is marked as "true" as well because the resolver ends. And we'd do so without any new API or low-level action. What's wrong with that?

Reply:
@youknowriad I may be wrong here, but the way I understand it is as follows:

*Short-circuiting in the resolver:*

This would go something like:

```js
export function* getEntityRecord( kind, name, key = '', query ) {
  // ...
  const record = yield findRecordInState( kind, name, key, query );
  if ( record ) { 
    return record;
  }
  // ...
  apiFetch()
  // ...
}
```

If I call `getEntityRecord` and the record already available, `apiFetch` won't be called - good. Now, imagine it's being invalidated by the following action:

```
{
	type: 'RECEIVE_ITEMS,
	kind,
	name,
	items: [ newRecord ],
	invalidateCache: true,
};
```

The resolver gets triggered again, and it short-circuits again. `apiFetch()` will never be called again. That's okay if there is an active `useSelect()` hook somewhere actively calling `getEntityRecords()` to fetch the entire list of records. But maybe there's not - I could imagine such `useSelect()` being used in a temporary component such as modal - once it's unmounted, there is nothing to re-fetch that record.

Reply:
@youknowriad what I'm proposing here is resolving things *once* when they become available. Any invalidation would trigger the resolver as expected.

Reply:
Indeed, you're right. I think we can try it in that case; I don't like the usage of low-level things not in the way they were meant to be used initially but it seems like the only way for now.

Reply:
That said, I think the current fix is not 100% accurate, it should only happen if we fetched all the fields (no _fields in the query)

Reply:
Good point! Hm but then if we use a limited set of fields in both `getEntityRecords()` and `getEntityRecord()`, it would still make sense to return that partial record. Which means the resolution is not just an entity-level concept but also/instead a field-level concept. That smells like a large refactor. Let's take a note and stick to the easy option of checking if there's no `_fields` in the query.

Reply:
Fortunately _fields is only about a subset of fields according to https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_fields . It would get even more interesting if it would be possible to request _additional_ fields that are not present by default.

Reply:
I created an issue here https://github.com/WordPress/gutenberg/issues/26629